### PR TITLE
Add XLSX Protection capabilities

### PR DESCRIPTION
**Pull request by freshleafmedia**
This PR adds support for Workbook and Worksheet protection to XLSX.

- [WorkbookProtection OOXML Spec](https://c-rex.net/samples/ooxml/e1/Part4/OOXML_P4_DOCX_workbookProtection_topic_ID0ECHK4.html?hl=workbookprotection)
- [WorkbookProtection MS Spec](https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.workbookprotection?view=openxml-3.0.1)
- [SheetProtection OOXML Spec](https://c-rex.net/samples/ooxml/e1/Part4/OOXML_P4_DOCX_sheetProtection_topic_ID0ENIH5.html)
- [SpeetProtection MS Spec](https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.sheetprotection?view=openxml-3.0.1)

### CODE REVIEW

File: src/Writer/XLSX/Options/SheetProtection.php [hunk 1]
```diff
@@ -0,0 +1,285 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSpout\Writer\XLSX\Options;
+
+final class SheetProtection
```
Comment (Slamdunk) [inline src/Writer/XLSX/Options/SheetProtection.php]:
```suggestion
final readonly class SheetProtection
```
1. Please mark the class as `readonly`
2. Make all the properties `public` so we can get rid of the setters/getters

File: src/Writer/XLSX/Options/WorkbookProtection.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSpout\Writer\XLSX\Options;
+
+final class WorkbookProtection
```
Comment (Slamdunk) [inline src/Writer/XLSX/Options/WorkbookProtection.php]:
```suggestion
final readonly class WorkbookProtection
```
1. Please mark the class as `readonly`
2. Make all the properties `public` so we can get rid of the setters/getters

File: src/Writer/XLSX/Options/WorkbookProtection.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        return $this;
+    }
+
+    public function getXml(): string
+    {
+        return '<workbookProtection'.$this->getSheetViewAttributes().'/>';
+    }
+
+    private function createPasswordHash(string $password): string
```
Comment (Slamdunk) [inline src/Writer/XLSX/Options/WorkbookProtection.php]: This seems copy-pasted from `SheetProtection`.
Please:

1. Move the hashing function into a separate class
2. Add a specific test with hardcoded password and hash result

**Review by Slamdunk — CHANGES_REQUESTED** [review]

**Comment by freshleafmedia** [comment]
@Slamdunk I have made the requested changes. :+1:

**Review by Slamdunk — APPROVED** [review]

**Pull request merged**